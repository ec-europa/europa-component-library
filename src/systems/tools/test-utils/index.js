const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
