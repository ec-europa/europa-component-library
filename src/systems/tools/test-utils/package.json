{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
