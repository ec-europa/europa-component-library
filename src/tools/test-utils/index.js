const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
