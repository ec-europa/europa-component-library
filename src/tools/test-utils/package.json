{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
