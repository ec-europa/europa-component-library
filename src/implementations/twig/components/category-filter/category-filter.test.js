import { merge, renderTwigFileAsNode } from '@ecl/test-utils';

// Import data for tests
import enData from '@ecl/specs-component-category-filter/demo/data';

describe('Category filter', () => {
  const template = '@ecl/category-filter/category-filter.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(enData)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(enData, {
        extra_classes: 'custom-button custom-button--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(enData, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Translated', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(frData)).resolves.toMatchSnapshot();
    });
  });
});
