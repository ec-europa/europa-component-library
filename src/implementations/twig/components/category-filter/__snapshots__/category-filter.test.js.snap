// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category filter Default renders correctly 1`] = `
<jest>
  <div
    aria-expanded="false"
    class="ecl-category-filter"
    data-ecl-auto-init="CategoryFilter"
    data-ecl-category-filter=""
  >
    <ul
      class="ecl-category-filter__list"
    >
      <li
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1"
          href="/example"
        >
          
        Item 1
        
        </a>
      </li>
      <li
        aria-expanded="false"
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1 ecl-category-filter__item--has-children"
          href="/example"
        >
          
        Item 2 label
          
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-category-filter__item-icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </a>
        <ul
          class="ecl-category-filter__list"
        >
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2 ecl-category-filter__item--has-children"
              href="/example"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--m ecl-icon--rotate-90 ecl-category-filter__item-icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#solid-arrow"
                />
              </svg>
                      Item 2.1
        
            </a>
            <ul
              class="ecl-category-filter__list"
            >
              <li
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3"
                  href="/example"
                >
                  
        Item 2.1.1
        
                </a>
              </li>
              <li
                aria-expanded="false"
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3 ecl-category-filter__item--has-children"
                  href="/example"
                >
                  <svg
                    aria-hidden="true"
                    class="ecl-icon ecl-icon--m ecl-icon--rotate-90 ecl-category-filter__item-icon"
                    focusable="false"
                  >
                    <use
                      xlink:href="/icons.svg#solid-arrow"
                    />
                  </svg>
                          Item 2.1.2
        
                </a>
                <ul
                  class="ecl-category-filter__list"
                >
                  <li
                    class="ecl-category-filter__list-item"
                  >
                    <a
                      class="ecl-category-filter__item ecl-category-filter__item--level-4"
                      href="/example"
                    >
                      
        Item 2.1.2 a
        
                    </a>
                  </li>
                  <li
                    class="ecl-category-filter__list-item"
                  >
                    <a
                      class="ecl-category-filter__item ecl-category-filter__item--level-4"
                      href="/example"
                    >
                      
        Item 2.1.2 b
        
                    </a>
                  </li>
                  <li
                    class="ecl-category-filter__list-item"
                  >
                    <a
                      class="ecl-category-filter__item ecl-category-filter__item--level-4"
                      href="/example"
                    >
                      
        Item 2.1.2 c
        
                    </a>
                  </li>
                </ul>
              </li>
              <li
                aria-expanded="false"
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3"
                  href="/example"
                >
                  
        Item 2.1.3
        
                </a>
              </li>
              <li
                aria-expanded="false"
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3"
                  href="/example"
                >
                  
        Item 2.1.4
        
                </a>
              </li>
            </ul>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.2
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.3
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.4
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.5
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.6
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.7
        
            </a>
          </li>
        </ul>
      </li>
      <li
        aria-expanded="false"
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1 ecl-category-filter__item--has-children"
          href="/example"
        >
          
        Item 3 with a very long label
          
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-category-filter__item-icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </a>
        <ul
          class="ecl-category-filter__list"
        >
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 3.1
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 3.2
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 3.3
        
            </a>
          </li>
        </ul>
      </li>
      <li
        aria-expanded="false"
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1 ecl-category-filter__item--has-children"
          href="/example"
        >
          
        Item 4 label
          
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-category-filter__item-icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </a>
        <ul
          class="ecl-category-filter__list"
        >
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.1
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.2
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.3 with a very long label going on 2 lines
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.4
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.5
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.6
        
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`Category filter Default renders correctly with extra attributes 1`] = `
<jest>
  <div
    aria-expanded="false"
    class="ecl-category-filter"
    data-ecl-auto-init="CategoryFilter"
    data-ecl-category-filter=""
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <ul
      class="ecl-category-filter__list"
    >
      <li
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1"
          href="/example"
        >
          
        Item 1
        
        </a>
      </li>
      <li
        aria-expanded="false"
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1 ecl-category-filter__item--has-children"
          href="/example"
        >
          
        Item 2 label
          
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-category-filter__item-icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </a>
        <ul
          class="ecl-category-filter__list"
        >
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2 ecl-category-filter__item--has-children"
              href="/example"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--m ecl-icon--rotate-90 ecl-category-filter__item-icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#solid-arrow"
                />
              </svg>
                      Item 2.1
        
            </a>
            <ul
              class="ecl-category-filter__list"
            >
              <li
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3"
                  href="/example"
                >
                  
        Item 2.1.1
        
                </a>
              </li>
              <li
                aria-expanded="false"
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3 ecl-category-filter__item--has-children"
                  href="/example"
                >
                  <svg
                    aria-hidden="true"
                    class="ecl-icon ecl-icon--m ecl-icon--rotate-90 ecl-category-filter__item-icon"
                    focusable="false"
                  >
                    <use
                      xlink:href="/icons.svg#solid-arrow"
                    />
                  </svg>
                          Item 2.1.2
        
                </a>
                <ul
                  class="ecl-category-filter__list"
                >
                  <li
                    class="ecl-category-filter__list-item"
                  >
                    <a
                      class="ecl-category-filter__item ecl-category-filter__item--level-4"
                      href="/example"
                    >
                      
        Item 2.1.2 a
        
                    </a>
                  </li>
                  <li
                    class="ecl-category-filter__list-item"
                  >
                    <a
                      class="ecl-category-filter__item ecl-category-filter__item--level-4"
                      href="/example"
                    >
                      
        Item 2.1.2 b
        
                    </a>
                  </li>
                  <li
                    class="ecl-category-filter__list-item"
                  >
                    <a
                      class="ecl-category-filter__item ecl-category-filter__item--level-4"
                      href="/example"
                    >
                      
        Item 2.1.2 c
        
                    </a>
                  </li>
                </ul>
              </li>
              <li
                aria-expanded="false"
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3"
                  href="/example"
                >
                  
        Item 2.1.3
        
                </a>
              </li>
              <li
                aria-expanded="false"
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3"
                  href="/example"
                >
                  
        Item 2.1.4
        
                </a>
              </li>
            </ul>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.2
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.3
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.4
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.5
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.6
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.7
        
            </a>
          </li>
        </ul>
      </li>
      <li
        aria-expanded="false"
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1 ecl-category-filter__item--has-children"
          href="/example"
        >
          
        Item 3 with a very long label
          
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-category-filter__item-icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </a>
        <ul
          class="ecl-category-filter__list"
        >
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 3.1
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 3.2
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 3.3
        
            </a>
          </li>
        </ul>
      </li>
      <li
        aria-expanded="false"
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1 ecl-category-filter__item--has-children"
          href="/example"
        >
          
        Item 4 label
          
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-category-filter__item-icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </a>
        <ul
          class="ecl-category-filter__list"
        >
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.1
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.2
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.3 with a very long label going on 2 lines
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.4
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.5
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.6
        
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`Category filter Default renders correctly with extra class names 1`] = `
<jest>
  <div
    aria-expanded="false"
    class="ecl-category-filter custom-button custom-button--test"
    data-ecl-auto-init="CategoryFilter"
    data-ecl-category-filter=""
  >
    <ul
      class="ecl-category-filter__list"
    >
      <li
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1"
          href="/example"
        >
          
        Item 1
        
        </a>
      </li>
      <li
        aria-expanded="false"
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1 ecl-category-filter__item--has-children"
          href="/example"
        >
          
        Item 2 label
          
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-category-filter__item-icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </a>
        <ul
          class="ecl-category-filter__list"
        >
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2 ecl-category-filter__item--has-children"
              href="/example"
            >
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--m ecl-icon--rotate-90 ecl-category-filter__item-icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#solid-arrow"
                />
              </svg>
                      Item 2.1
        
            </a>
            <ul
              class="ecl-category-filter__list"
            >
              <li
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3"
                  href="/example"
                >
                  
        Item 2.1.1
        
                </a>
              </li>
              <li
                aria-expanded="false"
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3 ecl-category-filter__item--has-children"
                  href="/example"
                >
                  <svg
                    aria-hidden="true"
                    class="ecl-icon ecl-icon--m ecl-icon--rotate-90 ecl-category-filter__item-icon"
                    focusable="false"
                  >
                    <use
                      xlink:href="/icons.svg#solid-arrow"
                    />
                  </svg>
                          Item 2.1.2
        
                </a>
                <ul
                  class="ecl-category-filter__list"
                >
                  <li
                    class="ecl-category-filter__list-item"
                  >
                    <a
                      class="ecl-category-filter__item ecl-category-filter__item--level-4"
                      href="/example"
                    >
                      
        Item 2.1.2 a
        
                    </a>
                  </li>
                  <li
                    class="ecl-category-filter__list-item"
                  >
                    <a
                      class="ecl-category-filter__item ecl-category-filter__item--level-4"
                      href="/example"
                    >
                      
        Item 2.1.2 b
        
                    </a>
                  </li>
                  <li
                    class="ecl-category-filter__list-item"
                  >
                    <a
                      class="ecl-category-filter__item ecl-category-filter__item--level-4"
                      href="/example"
                    >
                      
        Item 2.1.2 c
        
                    </a>
                  </li>
                </ul>
              </li>
              <li
                aria-expanded="false"
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3"
                  href="/example"
                >
                  
        Item 2.1.3
        
                </a>
              </li>
              <li
                aria-expanded="false"
                class="ecl-category-filter__list-item"
              >
                <a
                  class="ecl-category-filter__item ecl-category-filter__item--level-3"
                  href="/example"
                >
                  
        Item 2.1.4
        
                </a>
              </li>
            </ul>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.2
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.3
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.4
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.5
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.6
        
            </a>
          </li>
          <li
            aria-expanded="false"
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 2.7
        
            </a>
          </li>
        </ul>
      </li>
      <li
        aria-expanded="false"
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1 ecl-category-filter__item--has-children"
          href="/example"
        >
          
        Item 3 with a very long label
          
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-category-filter__item-icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </a>
        <ul
          class="ecl-category-filter__list"
        >
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 3.1
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 3.2
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 3.3
        
            </a>
          </li>
        </ul>
      </li>
      <li
        aria-expanded="false"
        class="ecl-category-filter__list-item"
      >
        <a
          class="ecl-category-filter__item ecl-category-filter__item--level-1 ecl-category-filter__item--has-children"
          href="/example"
        >
          
        Item 4 label
          
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-180 ecl-category-filter__item-icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#corner-arrow"
            />
          </svg>
        </a>
        <ul
          class="ecl-category-filter__list"
        >
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.1
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.2
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.3 with a very long label going on 2 lines
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.4
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.5
        
            </a>
          </li>
          <li
            class="ecl-category-filter__list-item"
          >
            <a
              class="ecl-category-filter__item ecl-category-filter__item--level-2"
              href="/example"
            >
              
        Item 4.6
        
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</jest>
`;
